### PR TITLE
S3 001

### PR DESCRIPTION
**Pull request by bcw1000**
Implementation of S3-001 remediation.

### CODE REVIEW

File: functions/AutoRemediateS3-001.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+      else     console.log(data);           // successful response
+    });
+
+    // actually close off access
+    var params = {
+      Bucket: event.resource,
+      ACL: "private",
+    };
+    s3.putBucketAcl(params, function(err, data) {
+      if (err) console.log(err, err.stack); // an error occurred
```
Comment (temyers) [inline functions/AutoRemediateS3-001.js]: If this call fails, then this lambda should fail?

`callback(err, "Failed to auto-remediate S3-001")`

**Comment by Hamo69** [comment]
Hi @bcw1000 what is happening with this?

**Comment by bcw1000** [comment]
I'm restoring this on my new machine and investigating.  I had committed the requested changes.

**Comment by bcw1000** [comment]
Hi Mike,

This version only removes READ permission.  In the case of FULL_CONTROL, it adds the remaining 3.
Let me know if you have any concerns.

Bret

**Comment by bcw1000** [comment]
Hi Mike,

This version only removes READ permission.  In the case of FULL_CONTROL, it adds the remaining 3.
Let me know if you have any concerns.

Bret

**Comment by bcw1000** [comment]
Hi Mike,

Tim points out there is a separate remediation warning for 'FULL_CONTROL'.  I'll remove that logic and re-submit.

Bret

**Comment by bcw1000** [comment]
Hi Mike,
I've removed the FULL_CONTROL code.  Let me know if you have any concerns.
Bret

File: serverless.yml [hunk 2]
```diff
@@ -91,7 +92,8 @@ provider:
 
     - Effect: Allow
       Action:
-        - redshift:ModifyCluster
```
Comment (mikerahmati) [inline serverless.yml]: Hi @bcw1000. This would break AutoRemediateRS-001 ?! Is there a reason to remove this ?

Comment (mikerahmati) [reply serverless.yml]: ^ Planning to re-design this part to have one IAM Role per Lambda function to a) avoid such issues b) implement the Principle Of Least Privilege and comply with https://www.cloudconformity.com/conformity-rules/Lambda/sharing-an-iam-role-within-more-than-one-lambda-function.html

File: serverless.yml [hunk 3]
```diff
@@ -91,7 +92,8 @@ provider:
...
+        - sns:ListTopics
+        - sns:Publish
```
Comment (mikerahmati) [inline serverless.yml]: Also, is there a reason for  sns:Publish & sns:ListTopics permission ?

Comment (bcw1000) [reply serverless.yml]: No.  A mistake in editing.

**Comment by bcw1000** [comment]
Hi Mike,
Addressed.  Please let me know.
Bret

**Pull request merged**